activerecord/lib/rails/generators/active_record/migration/templates/create_table_migration.rb.tt

```diff
@@ -6,6 +6,8 @@ class <%= migration_class_name %> < ActiveRecord::Migration[<%= ActiveRecord::Mi
       t.string :password_digest<%= attribute.inject_options %>
 <% elsif attribute.token? -%>
       t.string :<%= attribute.name %><%= attribute.inject_options %>
+<% elsif attribute.reference? -%>
+      t.references :<%= attribute.name %><%= attribute.inject_options %> <%= primary_key_type.nil? ? "" :  ",type: :#{primary_key_type}" %>
```

Comment:
This will generate a trailing space in case `primary_key_type` is `nil`. Also it is missing a space after the `,`.

We should also invert the conditional to avoid the `nil?` check.